notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 70))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data['family'] == value].groupby([\"date\"]).mean()\n",
+    "  sns.distplot(current_value_data['sales'], color='g', bins=100, hist_kws={'alpha': 0.4});\n",
+    "  plt.title(value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>These plots show that, roughly speaking, sales distributions of all families divided on two parts: <br>1) Normal or close to normal (such as 'automotive', 'bread/bakery', 'cleaning', 'eggs', 'grocery', 'lingerie', ...). Interesting fact that most of distributions from this category have right asymmetry (asymmetry coefficient is positive). The prove is below.<br>2) Distribution, where the biggest density is concentrated in zero or near zero. Other data is distributed differently (some values such as 'home and kitchen i', 'ladieswear' have something like normal distributions). It means that such goods categories aren't essential for people, that is why a number of sales during the day mostly is equal to 0.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for value in family_values:\n",
+    "    print(value, ': ', round(train_data[train_data['family'] == value].groupby([\"date\"]).mean().skew()['sales'], 2))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sales_and_onpromotion_data = train_data.drop(['date', 'store_nbr', 'family'], axis=1)\n",
+    "sales_and_onpromotion_data.corr()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data[\"family\"] == value].groupby([\"date\"]).mean()\n",
+    "  plt.scatter(x='onpromotion', y='sales', data=current_value_data)\n",
+    "  plt.title(value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>It becomes understandable that promotions have a pretty good influence on sales INCREASING (in general, talking about all the data).<br>As we can see on the plots above, most of the 'family' values (but not all of them!) prove this fact. Interesting fact that 'books' didn't have any promotions during he whole period of observations.<br>Moreover, correlation is influenced by outliers, so this coefficient may not be accurate.<br>Nevertheless, 'onpromotion' feature is useful for the predictions.</i>"
```

Comment:
I would suggest making transparent points in your graphs (play parameter `alpha`). Probably some points (combinations of promo + sales values) repeat many times, it is interesting to see, if that is true or not in your data.

> As we can see on the plots above, most of the 'family' values (but not all of them!) prove this fact.

Which categories do not follow the common trend and why?

> Moreover, correlation is influenced by outliers, so this coefficient may not be accurate.

Not sure I followed what you mean here. Can you please clarify this point?